client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcPreparedStatement.java

```diff
@@ -1051,128 +1074,225 @@ public void testConvertTimestamp()
+            throws Exception
     {
-        return new BindAssertion(this::createConnection, binder);
+        testLegacyPreparedStatementsSetting(false,
+                "EXECUTE IMMEDIATE 'SELECT * FROM blackhole.blackhole.test_table WHERE x = ? AND y = ? AND y <> ''Test''' USING INTEGER '42', 'value1''s'");
     }
 
-    private Connection createConnection()
+    private BindAssertion assertBind(Binder binder, boolean useLegacyPreparedStatements)
+    {
+        return new BindAssertion(() -> this.createConnection(useLegacyPreparedStatements), binder);
+    }
+
+    private Connection createConnection(boolean useLegacyPreparedStatements)
             throws SQLException
     {
         String url = format("jdbc:trino://%s", server.getAddress());
-        return DriverManager.getConnection(url, "test", null);
+        Properties properties = new Properties();
+        properties.setProperty("user", "test");
+        properties.setProperty("legacyPreparedStatements", String.valueOf(useLegacyPreparedStatements));
+        return DriverManager.getConnection(url, properties);
     }
 
-    private Connection createConnection(String catalog, String schema)
+    private Connection createConnection(String catalog, String schema, boolean useLegacyPreparedStatements)
             throws SQLException
     {
         String url = format("jdbc:trino://%s/%s/%s", server.getAddress(), catalog, schema);
-        return DriverManager.getConnection(url, "test", null);
+        Properties properties = new Properties();
+        properties.setProperty("user", "test");
+        properties.setProperty("legacyPreparedStatements", String.valueOf(useLegacyPreparedStatements));
+        return DriverManager.getConnection(url, properties);
+    }
+
+    private void testLegacyPreparedStatementsSetting(boolean legacyPreparedStatements, String expectedSql)
+            throws Exception
+    {
+        Method getExecuteSql = TrinoPreparedStatement.class.getDeclaredMethod("getExecuteSql", String.class, List.class);
+        getExecuteSql.setAccessible(true);
+        Method toValues = TrinoPreparedStatement.class.getDeclaredMethod("toValues", Map.class);
+        toValues.setAccessible(true);
+
+        Field parameters = TrinoPreparedStatement.class.getDeclaredField("parameters");
+        parameters.setAccessible(true);
+        Field batchValues = TrinoPreparedStatement.class.getDeclaredField("batchValues");
+        batchValues.setAccessible(true);
+
+        String selectSql = "SELECT * FROM blackhole.blackhole.test_table WHERE x = ? AND y = ? AND y <> 'Test'";
+        String insertSql = "INSERT INTO blackhole.blackhole.test_table (x, y) VALUES (?, ?)";
+
+        try (Connection connection = createConnection(legacyPreparedStatements)) {
+            try (Statement statement = connection.createStatement()) {
+                assertEquals(statement.executeUpdate("CREATE TABLE blackhole.blackhole.test_table (x bigint, y varchar)"), 0);
```

Comment:
There is no need to create a table if all you want is to make some assertions on the generated SQL.  Just select some values.

Reply:
If I don't create a table and look at the executed statements by querying `system.runtimes.queries` it works for the first scenario and testing the first batch. When calling `executeBatch()` only the first batch will be executed, an exception occurring before the second batch is run. But if we're fine testing only one batch then the whole thing can work.

Reply:
I suppose if you need a table if you want to test inserts.

Reply:
Tests updated